client/lib/parsing.rs

```diff
@@ -244,6 +244,44 @@ fn args_from_simple_or_complex(
     }
 }
 
+macro_rules! stringify_if_not_empty {
+    ($e: expr) => {
+        if !$e.is_empty() {
+            format!("{}={}", stringify!($e), $e)
+        } else {
+            "".to_string()
+        }
+    };
+}
+
+macro_rules! params_with_values {
+    ($($x:expr),+ $(,)?) => (vec![$((stringify_if_not_empty!($x))),+])
+}
+
+/// Private macro for enforcing that there are no conflicting arguments, including missing a
+/// required one. This is intended to make debugging the parse_*_info methods easier for callers.
+macro_rules! try_conflicting_params {
+    (context: $site: expr, $($x:expr),+ $(,)?) => {{
+        let required_arguments = params_with_values![$($x),+]
+            .iter()
+            .filter(|item| !item.is_empty())
+            .map(<_>::to_string)
+            .collect::<Vec<_>>();
+        if required_arguments.is_empty() {
+            return Err(Error::InvalidArgument(
+                $site,
+                "No arguments passed.".to_string(),
```

Comment:
This would be more useful if it said something like `Missing a required arg - exactly one of the following must be provided: {}` including the list of required args.

Don't worry about this though if it complicates the macro too much.